File: src/Other/Internals/ViewSourceButton.fs
namespace OpenSilver.Samples.Showcase

open System.Collections.Generic
open System.Windows
open System.Windows.Controls

type ViewSourceButton() as this =
    inherit Button()

    static let mutable showSourceCode : (ViewSourcePanel -> unit) = fun _ -> ()

    do
        this.Style <- Application.Current.Resources["ButtonViewSource_Style"] :?> Style

    static member ShowSourceCode
        with get() = showSourceCode
        and set v = showSourceCode <- v

    member val Sources = new List<ViewSourceButtonInfo>() with get

    override this.OnClick() =
        base.OnClick()
        ViewSourceButton.ViewSource(this.Sources)

    static member private ViewSource(sourcePaths: ICollection<ViewSourceButtonInfo>) =
        if sourcePaths = null || sourcePaths.Count = 0 then
            ()
        else
            let panel = new ViewSourcePanel()
            panel.ViewSource(sourcePaths)
            ViewSourceButton.ShowSourceCode(panel)

File: src/OpenSilver.Showcase/Other/Internals/ViewSourceButton.vb
Imports System.Windows
Imports System.Windows.Controls

Namespace Global.OpenSilver.Showcase
    Public NotInheritable Class ViewSourceButton
        Inherits Button

        Public Sub New()
            Style = TryCast(Application.Current.Resources("ButtonViewSource_Style"), Style)
        End Sub

        Public ReadOnly Property Sources As List(Of ViewSourceButtonInfo) = New List(Of ViewSourceButtonInfo)()

        Protected Overrides Sub OnClick()
            MyBase.OnClick()
            ViewSource(Sources)
        End Sub

        Private Shared Sub ViewSource(ByVal sourcePaths As ICollection(Of ViewSourceButtonInfo))
            If sourcePaths Is Nothing OrElse sourcePaths.Count = 0 Then
                Return
            End If

            Dim panel = New ViewSourcePanel()
            panel.ViewSource(sourcePaths)
            MainPage.Current.ViewSourceCode(panel)
        End Sub
    End Class
End Namespace
